src/shared/edit-util.c

```diff
@@ -312,39 +312,46 @@ static int strip_edit_temp_file(EditFile *e) {
         if (r < 0)
                 return log_error_errno(r, "Failed to read temporary file '%s': %m", e->temp);
 
+        tmp = strdup(old_contents);
+        if (!tmp)
+                return log_oom();
+
         if (e->context->marker_start) {
```

Comment:
The marker should not be used if we're in `stdin` mode.

Reply:
It's not. The markers are not initialized, but it's a bit hard to see in all the layers. I added an explicit assert now.

Reply:
Oh, they *are* initialized. I changed this yet again. PTAL.